Src/HALALMock/Services/EXTI/EXTI.cpp

```diff
@@ -26,25 +26,23 @@ uint8_t ExternalInterrupt::inscribe(Pin& pin, function<void()>&& action, TRIGGER
 		ErrorHandler(" The pin %s is already used or isn t available for EXTI usage", pin.to_string().c_str());
 		return 0;
 	}
-
-	if (trigger == RISING) {
-		Pin::inscribe(pin, EXTERNAL_INTERRUPT_RISING);
-	}
-	else if (trigger == FALLING) {
-		Pin::inscribe(pin, EXTERNAL_INTERRUPT_FALLING);
+	EmulatedPin &pin_data = SharedMemory::get_pin(pin);
+	if(pin_data.type != PinType::EXTIPin) {
+		ErrorHandler("ID %d is not registered as a EXTIPin",id);
+		return;
 	}
 
+	(pin_data.PinData.EXTIPin.trigger_mode) = trigger;
+
 	service_ids[id_counter] = pin;
 	instances[pin.gpio_pin].action = action;
 
 	return id_counter++;
 }
-
+//TODO: assigne priority on the emulated pin
```

Comment:
Don't worry too about priority of interrupts for now, we dont use them in the STLIB, (although we should) .

In case we want to actually use priorities we could make a global queue of signals